doc/audit/security_auditing_HLD.md

```diff
@@ -0,0 +1,513 @@
+  - [5. Q\&A](#5-qa)
+  - [6. Future Improvements](#6-future-improvements)
+
+## List of Tables
+* [Table 1: Revision](#table-1-revision)
+* [Table 2: Audit Rules Review](#table-2-audit-rules-review)
+* [Table 3: Unit Test Cases](#table-3-unit-test-cases)
+* [Table 4: System Test Cases](#table-4-system-test-cases)
+
+## Revision
+###### Table 1: Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 08/20/2024  | Mai Bui            | Initial version                   |
+
+## Scope
+This section describes the audit enhancement high-level design in SONiC.
+
+## 1. Overview
+This design aims to enhance the auditing capabilities within SONiC operating system using audit daemon (auditd). Auditing is the process of recording and analyzing the events that occur on the device. Auditing can help to detect unauthorized access, configuration changes, malicious activity, or system errors. Auditing can also provide evidence for forensic investigations, compliance audits, or incident response.
+
+This document outlines the addition of several auditd rules to enhance the auditing capabilities on a SONiC device. These extra rules aim to provide thorough monitoring of important system activities, making the device more secure and better prepared for compliance and incident response.
+
+## 2. Current Architecture Design
+In SONiC, audit settings are centrally managed through a configuration file at `/etc/audit/auditd.conf`. Upon system startup, SONiC automatically compiles audit rules from the `/etc/audit/rules.d/` directory into a single file, `/etc/audit/audit.rules`, which auditd then loads. Here is a brief look at the relevant files:
+- `/etc/audit/auditd.conf` - Main configuration file.
+- `/etc/audit/audit.rules` - Loaded at startup, compiled from the individual rules in the rules.d directory.
+- `/etc/audit/rules.d/` - Directory for individual rule files
+  - `audisp-tacplus.rules`
+  - `audit.rules`
+
+## 3. High-level Design
+### 3.1 Design
+- Introduce a new file, `security-auditing.rules`, into the `/etc/audit/rules.d/` directory
+- A predefined set of rules (detailed in section 3.2) will be automatically enabled as the default configuration.
+- Modify the `/etc/audit/plugins.d/syslog.conf` file by setting `active = yes` to enable the forwarding of auditd logs to a syslog server.
+- ConfigDB schema design, new AUDIT table in Config DB
+- YANG model
+- CLI commands to enable or disable all rules, with support for adding or removing individual rules for fine-grained control:
+  - `show audit`
+  - `config audit enable`
+  - `config audit disable`
+  - `config audit add <rule>`
+  - `config audit remove <rule>`
+- Logrotate
+- Audit rules ordering
+
+### 3.2 Audit Rules Review
+###### Table 2: Audit Rules Review
+These rules will be included in the image and enabled by default.
+| Rule name | Details |
```

Comment:
Could we add a column by providing examples of the audit rules?

Reply:
I added some audit logs examples, it looks similar, so I did not add entire column